aten/src/ATen/mps/MPSDevice.mm

```diff
@@ -119,6 +119,11 @@ static inline MTLLanguageVersion getMetalLanguageVersion(const id<MTLDevice>& de
 
   static bool _macos_14_0_plus = [mpsCD instancesRespondToSelector:@selector(conjugateWithTensor:name:)] == YES;
 
+  // TODO: Change all version checks to use this API
+  NSProcessInfo* processInfo = [[NSProcessInfo alloc] init];
+  static bool _macos_14_4_plus =
+      [processInfo isOperatingSystemAtLeastVersion:{.majorVersion = 14, .minorVersion = 4, .patchVersion = 0}];
```

Comment:
Nit (this way NSProcessInfo will be allocated only once, rather than on every call of the function
```suggestion
  static bool _macos_14_4_plus = (){
    NSProcessInfo* processInfo = [[NSProcessInfo alloc] init];
    return [processInfo isOperatingSystemAtLeastVersion:{.majorVersion = 14, .minorVersion = 4, .patchVersion = 0}];
     }();
```